homeassistant/components/template/weather.py

```diff
@@ -0,0 +1,203 @@
+"""Template platform that aggregates meteorological data."""
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_EXCEPTIONAL,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_HAIL,
+    ATTR_CONDITION_LIGHTNING,
+    ATTR_CONDITION_LIGHTNING_RAINY,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SNOWY_RAINY,
+    ATTR_CONDITION_SUNNY,
+    ATTR_CONDITION_WINDY,
+    ATTR_CONDITION_WINDY_VARIANT,
+    ENTITY_ID_FORMAT,
+    WeatherEntity,
+)
+from homeassistant.const import CONF_NAME, CONF_UNIQUE_ID
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.reload import async_setup_reload_service
+
+from .const import DOMAIN, PLATFORMS
+from .template_entity import TemplateEntity
+
+CONDITION_CLASSES = {
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_HAIL,
+    ATTR_CONDITION_LIGHTNING,
+    ATTR_CONDITION_LIGHTNING_RAINY,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SNOWY_RAINY,
+    ATTR_CONDITION_SUNNY,
+    ATTR_CONDITION_WINDY,
+    ATTR_CONDITION_WINDY_VARIANT,
+    ATTR_CONDITION_EXCEPTIONAL,
+}
+
+CONF_WEATHER = "weather"
+CONF_TEMPERATURE_TEMPLATE = "temperature_template"
+CONF_HUMIDITY_TEMPLATE = "humidity_template"
+CONF_CONDITION_TEMPLATE = "condition_template"
+CONF_PRESSURE_TEMPLATE = "pressure_template"
+CONF_WIND_SPEED_TEMPLATE = "wind_speed_template"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_CONDITION_TEMPLATE): cv.template,
+        vol.Required(CONF_TEMPERATURE_TEMPLATE): cv.template,
+        vol.Required(CONF_HUMIDITY_TEMPLATE): cv.template,
```

Comment:
The weather entity only requires condition and temperature according to the dev docs [here](https://developers.home-assistant.io/docs/core/entity/weather).  Is there a reason that Humidity is required here?

Reply:
just legacy from my belly button configuration (fixed)

Reply:
https://github.com/home-assistant/architecture/issues/485